src/rust/engine/src/scheduler.rs

```diff
@@ -291,11 +292,33 @@ impl Scheduler {
       .map(|s| s.into())
       .collect();
 
+    // This map keeps the k most relevant jobs in assigned possitions.
+    // Keys are positions in the display (display workers) and the values are the actual jobs to print.
+    let mut tasks_to_display = IndexMap::new();
+
     let results = loop {
       if let Ok(res) = receiver.recv_timeout(Duration::from_millis(100)) {
         break res;
       } else if let Some(display) = optional_display.as_mut() {
-        Scheduler::display_ongoing_tasks(&self.core.graph, &roots, display);
+        // Update the graph. To do that, we iterate over heavy hitters.
```

Comment:
Can we move this out of this method to keep the loop tiny and readable? It looks like it should be possible to do that by passing a `&mut IndexMap` into `display_ongoing_tasks`, and doing the extra calculations there.